### ISSUE TITLE
在回归任务中，多次预测，结果不一致

### ISSUE THREAD

**Issue by JankinTian** [microsoft/LightGBM#6415]
## Description
在使用lightGBM进行回归预测训练，并保存权重后，重新加载权重进行预测，多次启动预测程序，预测结果不一致。
在训练过程中固定了，随机种子（seed=42），
测试过程中，保证文件的顺序一致

## Reproducible example
train code：
```python
lgb_model = lgb.LGBMRegressor(
        task = 'train',
        objective = 'quantile',
        alpha = 0.85,
        boosting_type = 'gbdt',
        learning_rate = 0.003,
        n_estimators = 2000, 
        min_child_samples = 16, 
        max_depth = 14,  
        num_leaves = 128,
        seed = 42,
        random_state = 42,
        max_cat_threshold = 1024,
        max_bin = 256,
    )
    lgb_model.fit(X_train, y_train, eval_set = [(X_train, y_train), 
                                         (X_test, y_test)], 
                                        eval_metric = {'huber','l2'}, 
                                        verbose = 5,  
                                        early_stopping_rounds = 20)
```
predict code
```python
lgb_model = lgb.Booster(model_file = './model.txt')
lgb_model.predict(val_data, num_threads = 1, num_iteration=lgb_model.best_iteration)
```

## Environment info

```shell
pip install lightgbm=3.3.5
```
